disnake/ext/commands/interaction_bot_base.py

```diff
@@ -1201,26 +1279,47 @@ async def process_application_commands(
         interaction: :class:`disnake.ApplicationCommandInteraction`
             The interaction to process commands for.
         """
-        if self._sync_commands and not self._sync_queued:
+        if self._command_sync_flags._sync_enabled and not self._sync_queued:
             known_command = self.get_global_command(interaction.data.id)  # type: ignore
 
             if known_command is None:
                 known_command = self.get_guild_command(interaction.guild_id, interaction.data.id)  # type: ignore
 
             if known_command is None:
+                # don't do anything if we aren't allowed to disable them
                 # This usually comes from the blind spots of the sync algorithm.
                 # Since not all guild commands are cached, it is possible to experience such issues.
                 # In this case, the blind spot is the interaction guild, let's fix it:
-                try:
-                    await self.bulk_overwrite_guild_commands(interaction.guild_id, [])  # type: ignore
-                except disnake.HTTPException:
-                    pass
+                if self._command_sync_flags.allow_command_deletion:
```

Comment:
This doesn't fully fix #265 unless `allow_command_deletion` is disabled:
with `sync_global=True` and `sync_guild=False`, invoking a guild command (that may have been created from a web interface or similar, like in the linked issue) will result in the bot clearing all commands in that guild.

Reply:
https://github.com/DisnakeDev/disnake/pull/806/commits/4d3c7e48c74dab2e95627ef869a7157ffebf4fbd

Reply:
and https://github.com/DisnakeDev/disnake/pull/806/commits/bd6e394b32d31b66af166913c15124bff85748cb :^)